contracts/Ovm_SpokePool.sol

```diff
@@ -104,66 +104,17 @@ contract Ovm_SpokePool is SpokePool {
     }
 
     /**************************************
-     *         DATA WORKER FUNCTIONS      *
+     *        INTERNAL FUNCTIONS          *
      **************************************/
 
     /**
-     * @notice Wraps any ETH into WETH before executing base function. This is necessary because SpokePool receives
+     * @notice Wraps any ETH into WETH before executing leaves. This is necessary because SpokePool receives
      * ETH over the canonical token bridge instead of WETH.
-     * @inheritdoc SpokePool
      */
-    function executeSlowRelayLeaf(
-        address depositor,
-        address recipient,
-        address destinationToken,
-        uint256 totalRelayAmount,
-        uint256 originChainId,
-        int64 realizedLpFeePct,
-        int64 relayerFeePct,
-        uint32 depositId,
-        uint32 rootBundleId,
-        bytes memory message,
-        int256 payoutAdjustment,
-        bytes32[] memory proof
-    ) public override(SpokePool) nonReentrant {
-        if (destinationToken == address(wrappedNativeToken)) _depositEthToWeth();
-
-        _executeSlowRelayLeaf(
-            depositor,
-            recipient,
-            destinationToken,
-            totalRelayAmount,
-            originChainId,
-            chainId(),
-            realizedLpFeePct,
-            relayerFeePct,
-            depositId,
-            rootBundleId,
-            message,
-            payoutAdjustment,
-            proof
-        );
+    function _preExecuteLeafHook(address l2TokenAddress) internal override {
```

Comment:
Should we also make this virtual since it can be inherited by the downstream chain-specific contracts ?

Reply:
I think we can get to that if we see a need to but its not needed now and would mean extra override declarations in the child contracts